Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -406,6 +409,25 @@ private class TopSitesDataSource: NSObject, UICollectionViewDataSource {
         cell.imageView.contentMode = UIViewContentMode.Center
     }
 
+    private func setBlurredBackground(image: UIImage, withURL url: NSURL, forCell cell: ThumbnailCell) {
+        let blurredKey = "\(url.absoluteString)!blurred"
+        if let blurredImage = SDImageCache.sharedImageCache().imageFromMemoryCacheForKey(blurredKey) {
+            cell.backgroundImage.image = blurredImage
+        } else {
+            dispatch_async(self.blurQueue) {
+                let blurredImage = image.applyLightEffect()
```

Comment:
This PR makes only a slight difference but after the query timing, the second highest cost is this blur. If we're unsure about the stability of this, I can look into fixing some of the layout issues as well to see if we can get away with just doing layout fixes.